WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -1101,8 +1104,30 @@ private extension ProductFormViewController {
         guard let site = ServiceLocator.stores.sessionManager.defaultSite else {
             return
         }
-        let viewModel = BlazeWebViewModel(source: .productMoreMenu, siteURL: site.url, productID: product.productID)
-        let webViewController = AuthenticatedWebViewController(viewModel: viewModel)
+        viewModel.checkIfBlazeIntroViewIsNeeded()
+
+        if viewModel.shouldShowBlazeIntroView {
+            let blazeCampaignIntroView = BlazeCampaignIntroView(onStartCampaign: {
```

Comment:
We should avoid capture `self` strongly in escaping closures to avoid retain cycles. This [article](https://www.avanderlee.com/swift/weak-self/) can be helpful for giving a quick look into how we want to work with `self` in closures.

What we usually do in WCiOS is capturing self weakly and use a `guard` to get a non-optional `self` if needed:

```suggestion
            let blazeCampaignIntroView = BlazeCampaignIntroView(onStartCampaign: { [weak self] in
                guard let self else { return }
```